mono/utils/mono-state.c

```diff
@@ -116,8 +116,13 @@ mono_summarize_toggle_assertions (gboolean enable)
 	}
 
 	mono_memory_barrier ();
-#endif
 }
```

Comment:
What's the reason behind this change?  I think it looks more ugly when we have the same function defined both inside and outside the conditional.

Reply:
The reason is to move unused *_not_reached_* functions under conditional block. I was thinking one block is better than two for the same condition, but surely I can keep function intact if you prefer.

Reply:
I agree with @baulig here, we use that style everywhere and even you do it in `mono_native_state_add_ee_info` below 🙂

Reply:
Sure, I'll keep that in mind. To me the difference was about having conditional expression in one place or in two, having it consolidated seemed better.